.github/workflows/pylint.yml

```diff
@@ -0,0 +1,23 @@
+name: Pylint
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install pylint
+    - name: Analysing the code with pylint
+      run: |
+        pylint $(git ls-files '*.py')
```

Comment:
The code base likely has a lot of pylint violations, to enable for developers it will have to be able to generate a list of modified files for the given PR and run the linter only against the touched files

Reply:
Thanks @eyurtsev for reviewing the PR and the suggestions. I'll look into it and modify it accordingly, Thanks